src/main/java/seedu/address/ui/ResultDisplay.java

```diff
@@ -12,12 +15,25 @@
 public class ResultDisplay extends UiPart<Region> {
 
     private static final String FXML = "ResultDisplay.fxml";
+    private final Logger resultLogger = Logger.getLogger(ResultDisplay.class.getName());
 
     @FXML
     private TextArea resultDisplay;
 
+    /**
+     * Creates a {@code ResultDisplay}.
+     */
     public ResultDisplay() {
         super(FXML);
+        resultDisplay.textProperty().addListener((obs, oldText, newText) -> {
+            resultDisplay.setPrefHeight((newText.chars().filter(num -> num == '\n').count() + 1)
+                    * resultDisplay.getFont().getSize() * 1.5 + 5);
+            resultDisplay.setMinHeight((newText.chars().filter(num -> num == '\n').count() + 1)
+                    * resultDisplay.getFont().getSize() * 1.5 + 5);
+            resultLogger.log(Level.INFO, "Setting Height of ResultDisplay to "
+                    + (newText.chars().filter(num -> num == '\n').count() + 1)
+                        * resultDisplay.getFont().getSize() * 1.5 + 5);
```

Comment:
likewise, there are name "magic numbers" such as 1.5 and 5 here can consider naming it as offset or scale?